client/state/domains/actions.js

```diff
@@ -20,6 +20,19 @@ export const recordAddDomainButtonClick = ( domainName, section ) => composeAnal
 	} ),
 );
 
+export const recordAddDomainButtonClickInMapDomain = ( domainName, section ) => composeAnalytics(
```

Comment:
I don't _love_ that this action creator is named with a one-time-use context (`InMapDomain` === not re-usable) but it's definitely hard to avoid this since the events are so specific to the context. I think this is probably fine for now, so not a blocker.